Client/Frontend/Browser/Playlist/Controllers/PlaylistListViewController.swift

```diff
@@ -247,20 +255,37 @@ class PlaylistListViewController: UIViewController {
     }
     
     // Shared folders are loading
-    if PlaylistManager.shared.currentFolder == nil {
+    if PlaylistManager.shared.currentFolder?.isPersistent == false {
       autoPlayEnabled = true
       return
     }
 
     // Setup initial playback item and time-offset
+    var lastPlayedItemTime: Double = 0.0
+    
     let lastPlayedItemUrl = initialItem?.pageSrc ?? Preferences.Playlist.lastPlayedItemUrl.value
-    let lastPlayedItemTime = initialItem != nil ? initialItemOffset : Preferences.Playlist.lastPlayedItemTime.value
+    let lastPlayedItemId = PlaylistManager.shared.allItems.first(where: { $0.pageSrc == lastPlayedItemUrl })?.tagId
+    
+    // If the user is current viewing the same video as the last played item
+    // then we choose whichever time offset is more recent
+    if let pageSrc = initialItem?.pageSrc,
+       let lastPlayedItem = Preferences.Playlist.lastPlayedItemUrl.value,
+       pageSrc == lastPlayedItem {
+      
+      // User is current on the same page as the last played item
+      lastPlayedItemTime = initialItemOffset > Preferences.Playlist.lastPlayedItemTime.value ?
```

Comment:
nit: I believe we can use `max` value here between `initialItemOffset` and `Preferences.Playlist.lastPlayedItemTime.value`